subsys/net/lib/http/http_server_core.c

```diff
@@ -873,7 +873,21 @@ int http_server_sendall(struct http_client_ctx *client, const void *buf, size_t
 	while (len) {
 		ssize_t out_len = zsock_send(client->fd, buf, len, 0);
 
-		if (out_len < 0) {
+		if ((out_len == 0) || (out_len < 0 && errno == EAGAIN)) {
+			struct zsock_pollfd pfd;
+			int pollres;
+
+			pfd.fd = sock;
+			pfd.events = ZSOCK_POLLOUT;
+			pollres = zsock_poll(&pfd, 1, 0);
```

Comment:
Setting timeout as 0, which means no timeout, is probably something we do not want to do here. What about introducing timeout to this function like in http-client case?

Reply:
@jukkar Yes, it would be probably better. Should a new constant be introduced or some existing one reused?